internal/spdxlicense/license.go

```diff
@@ -1,12 +1,38 @@
 package spdxlicense
 
 import (
+	"regexp"
 	"strings"
 )
 
+// https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#license-short-name
+// If a license generated in license_list.go is not found when looking up by ID,
+// then the ID function will use these regular expressions to help resolve cases where
+// x.0.0 and x are supplied as version numbers. For SPDX compatibility, versions with trailing
+// dot-zeroes are considered to be equivalent to versions without (e.g., “2.0.0” is considered equal to “2.0” and “2”).
+// EX: gpl-2+ ---> GPL-2.0+
+// EX: gpl-2.0.0-only ---> GPL-2.0-only
+// See the debian link for more details on the spdx license differences
+var (
+	zero   = regexp.MustCompile(`^((.*).0)(.*)$`)
+	noZero = regexp.MustCompile(`^(.*-)([1-9])(.*)`)
+)
+
 //go:generate go run generate_license_list.go
 
 func ID(id string) (string, bool) {
```

Comment:
this approach is changing the input license at runtime, but we have an opportunity to generate all possible answers in the lookup at build time (over in `generate_license_list.go`)

Reply:
If we move if over into generation should we be explicit in the capture groups for only debian short license:
https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#license-short-name

Or do we want our map to just add cases `x-1` and `x-1.0.0`

How high do we want the version numbers to go in the generation?

Reply:
Since we are iterating over this list I think it's pretty simple to add `2` extra key given the example
```
	"gpl-1.0-only":                         "GPL-1.0-only",
	"gpl-1.0-or-later":                     "GPL-1.0-or-later",
	"gpl-2.0":                              "GPL-2.0",
	"gpl-2.0+":                             "GPL-2.0+",
	"gpl-2.0-only":                         "GPL-2.0-only",
	"gpl-2.0-or-later":                     "GPL-2.0-or-later",
	"gpl-2.0-with-autoconf-exception":      "GPL-2.0-with-autoconf-exception",
	"gpl-2.0-with-bison-exception":         "GPL-2.0-with-bison-exception",
	"gpl-2.0-with-classpath-exception":     "GPL-2.0-with-classpath-exception",
	"gpl-2.0-with-font-exception":          "GPL-2.0-with-font-exception",
	"gpl-2.0-with-gcc-exception":           "GPL-2.0-with-GCC-exception",
	"gpl-3.0":                              "GPL-3.0",
	"gpl-3.0+":                             "GPL-3.0+",
	"gpl-3.0-only":                         "GPL-3.0-only",
	"gpl-3.0-or-later":                     "GPL-3.0-or-later",
	"gpl-3.0-with-autoconf-exception":      "GPL-3.0-with-autoconf-exception",
	"gpl-3.0-with-gcc-exception":           "GPL-3.0-with-GCC-exception",
```

So `gpl-3.0-only` becomes `gpl-3.0-only`, `gpl-3-only` `gpl-3.0.0-only` and they all point to the same `GPL-3.0-only`

Reply:
@wagoodman bumped the code to generate the map at build time with entries. 

Tests I wrote for runtime still work which says to me 👍